compiler/cgen.nim

```
@@ -1630,6 +1630,9 @@ proc cgenWriteModules*(backend: RootRef, config: ConfigRef) =
   # deps are allowed (and the system module is processed in the wrong
   # order anyway)
   g.config = config
+  let (outDir, _, _) = splitFile(config.outfile)
+  if not outDir.isEmpty:
+    try: createDir(outDir) except: discard "couldn't create dir"
```

Let the exception bubble up, don't hide it.